2006-12-28  David S. Miller  <davem@davemloft.net>

	* shlib-versions: Fix sparc64 linux target specification.

2006-10-02  Jakub Jelinek  <jakub@redhat.com>

	[BZ #3291]
	* sysdeps/unix/sysv/linux/sparc/sparc64/pause.c: Include
	errno.h, signal.h, unistd.h and sysdep-cancel.h.
	(__sigprocmask): Define.

--- glibc-2.5/sysdeps/unix/sysv/linux/sparc/sparc64/pause.c.orig	2006-08-15 05:27:23.000000000 +0000
+++ glibc-2.5/sysdeps/unix/sysv/linux/sparc/sparc64/pause.c	2006-12-31 15:12:41.000000000 +0000
@@ -1 +1,9 @@
+#include <errno.h>
+#include <signal.h>
+#include <unistd.h>
+#include <sysdep-cancel.h>
+
+#define __sigprocmask(how, set, oset) \
+  INLINE_SYSCALL (rt_sigprocmask, 4, how, set, oset, _NSIG / 8)
+
 #include <sysdeps/posix/pause.c>
--- glibc-2.5/nptl/shlib-versions.orig	2004-07-05 17:03:27.000000000 +0000
+++ glibc-2.5/nptl/shlib-versions	2006-12-31 15:20:09.000000000 +0000
@@ -1,5 +1,5 @@
 mips.*-.*-linux.*	libpthread=0		GLIBC_2.0 GLIBC_2.2
-sparc64-.*-linux.*	libpthread=0		GLIBC_2.2
+sparc64.*-.*-linux.*	libpthread=0		GLIBC_2.2
 sh.*-.*-linux.*		libpthread=0		GLIBC_2.2
 ia64.*-.*-linux.*	libpthread=0		GLIBC_2.2
 hppa.*-.*-linux.*	libpthread=0		GLIBC_2.2
